underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java

```diff
@@ -154,7 +155,13 @@ public OutputStream createDirect(String path, CreateOptions options) throws IOEx
       try {
         LOG.debug("Creating HDFS file at {} with perm {}", path, perm.toString());
         // TODO(chaomin): support creating HDFS files with specified block size and replication.
-        return FileSystem.create(mFileSystem, new Path(path),
+        Path hdfsPath = new Path(path);
+        if (!options.getCreateParent()) {
```

Comment:
Yes, FileSystem.create will automatically create parent directories. HDFS has a call called createNonRecursive which is what we need. However, it was deprecated and then un-deprecated in 3.0.0-alpha1: http://hadoop.apache.org/docs/r3.0.0-alpha1/api/org/apache/hadoop/fs/FileSystem.html.

For HDFS vendors who have not implemented createNonRecursive, we could try non-recursive first and then fallback to the code above if an IOException is thrown. @calvinjia thoughts?.